irods/test/meta_test.py

```diff
@@ -121,6 +122,18 @@ def tearDown(self):
 
     from irods.test.helpers import create_simple_resc_hierarchy
 
+    def test_atomic_metadata_json_error__issue_606(self):
+        path = self.coll_path + "/atomic_meta_issue_606"
+        obj = self.sess.data_objects.create(path)
+        obj.unlink(force = True)
+        try:
+            obj.metadata.apply_atomic_operations(AVUOperation(operation="add", avu=iRODSMeta('a','b','c')))
+        except ex.iRODSException as e:
+            resp = e.server_msg.get_json_encoded_struct()
```

Comment:
Does this depend on the API invoked?
Can this fail? In what way?

Reply:
It could fail if the API's output is other than json formatted.  Otherwise, as I've noted in the separate discussion below, APIs with char** or BytesBuf_t** or BinBytesBuf_t** as output, it should work.  Let's resume any discussion of how it could fail in the below discussion and close this one for now.